### PR TITLE
feat: Add hotjar

### PR DESCRIPTION
**Pull request by danielpartida**
## What it solves

UX team would like to have heatmaps, i.e. see what users click, what users are missing from our landing page

## How this PR fixes it
Please test the implementation in the staging environment -> https://safe-web-landing.staging.5afe.dev/

* Hotjar script is added/removed from the page following analytics cookies preferences.
* New Hotjar section in the Privacy Notice page
* Hotjar update on the Cookie Policy, including all the potential cookies the software can set.

Documentation of Hotjar [here](https://www.notion.so/safe-global/Set-up-Hotjar-b3b848371f9a4ac7bbd21e7f88b5f42f)

### CODE REVIEW

**Comment by katspaugh** [comment]
@danielpartida I've changed the script tag a bit for it to be React-compatible.

File: src/pages/_document.tsx [hunk 1]
```diff
@@ -39,8 +40,14 @@ export default class MyDocument extends Document {
           <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon/favicon-32x32.png" />
           <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon/favicon-16x16.png" />
 
-          {/* Plausible Analytics */}
-          <script defer data-domain="safe.global" src="https://plausible.io/js/script.js"></script>
+          {IS_PRODUCTION && (
```
Comment (iamacook) [inline src/pages/_document.tsx:43]:
```suggestion
          {IS_PRODUCTION && IS_OFFICIAL_HOST && (
```

Need we also add this to the `ContentSecurityPolicy`?

Comment (katspaugh) [reply src/pages/_document.tsx:43]: AFAICS, we have neither this variable nor CSP in this repo.

Comment (iamacook) [reply src/pages/_document.tsx:43]: Very true, I thought we were in another one 🙈

**Review by usame-algan — CHANGES_REQUESTED** [review]
We need to make sure that the Hotjar script is only included if the analytics cookie is set / was accepted

File: src/pages/_document.tsx [hunk 2]
```diff
@@ -40,8 +40,12 @@ export default class MyDocument extends Document {
           <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon/favicon-32x32.png" />
           <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon/favicon-16x16.png" />
 
-          {/* Plausible Analytics */}
-          {IS_PRODUCTION && <script defer data-domain="safe.global" src="https://plausible.io/js/script.js" />}
+          {IS_PRODUCTION && (
+            <>
+              {/* Plausible Analytics */}
+              <script defer data-domain="safe.global" src="https://plausible.io/js/script.js" />
```
Comment (katspaugh) [inline src/pages/_document.tsx:46]: Could you also remove this? We decided not to go with it.
cc @danielpartida

Comment (DiogoSoaress) [reply src/pages/_document.tsx]: removed

**Review by katspaugh — CHANGES_REQUESTED** [review]
I cannot test it because it's only enabled if IS_PRODUCTION. Please remove the flag temporarily.

Also I assume Hotjar uses cookies (since we ask for a cookie consent to enable it) but the Cookie Policy only mentions GA.

**Comment by DiogoSoaress** [comment]
> I cannot test it because it's only enabled if IS_PRODUCTION. Please remove the flag temporarily.

I am temporarily modifying the `deploy` workflow to deploy this branch to the staging environment. Reason being it is the corresponding domain of `NEXT_PUBLIC_HOTJAR_ID_STAGING`.

**Comment by katspaugh** [comment]
I still don't see hotjar loaded. You need to pass the new secrets to the build here: https://github.com/safe-global/safe-homepage/blob/main/.github/workflows/build/action.yml#L20-L21

Please don't ask me for a review until you have verified yourself that it works.

**Comment by DiogoSoaress** [comment]
I will block this PR until we have assessed the Cookie Policy update.

File: src/hooks/useHotjar.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+const ID = IS_PRODUCTION ? HOTJAR_ID : HOTJAR_ID_STAGING
+
+const useHotjar = () => {
+  const { isAnalyticsEnabled } = useCookieBannerContext()
+
+  // Add/remove hotjar script
+  useEffect(() => {
+    if (isAnalyticsEnabled) {
+      scriptRef = document.createElement('script')
+      scriptRef.innerText = `(function(h,o,t,j,a,r){h.hj=h.hj||function(){(h.hj.q=h.hj.q||[]).push(arguments)};h._hjSettings={hjid:${ID},hjsv:${HOTJAR_VERSION}};a=o.getElementsByTagName('head')[0];r=o.createElement('script');r.async=1;r.src=t+h._hjSettings.hjid+j+h._hjSettings.hjsv;a.appendChild(r);})(window,document,'https://static.hotjar.com/c/hotjar-','.js?sv=');`
```
Comment (DiogoSoaress) [inline src/hooks/useHotjar.ts:16]: I had to set the `script.innerText` so the installation could be verified on Hotjar's website

**Comment by DiogoSoaress** [comment]
The changes in `.github/workflows/deploy.yml` will be reverted before merging the PR.
These changes currently serve the purpose of verifying the successful installation of Hotjar in `https://safe-web-landing.staging.5afe.dev/`

**Comment by katspaugh** [comment]
The hotjar script is loaded when I accept analytics cookies but I don't see any events sent to hotjar afterwards.

Also there's an error in the console:

Screenshot 2023-08-29 at 10 54 55

**Comment by DiogoSoaress** [comment]
> Also there's an error in the console

Are you testing in https://safe-web-landing.staging.5afe.dev/ ??

> The hotjar script is loaded when I accept analytics cookies but I don't see any events sent to hotjar afterwards.

I just set up a Heatmap and it was successfully recorded. I also see Hotjar populating my Cookie storage
Screenshot 2023-08-29 at 13 12 04

**Comment by katspaugh** [comment]
I see no errors now and a working websocket connection. ✅

**Review by katspaugh — APPROVED** [review]
LGTM.

Don't forget to restore deploy.yml. On that note, you might want to introduce a third permanent branch (e.g. `feature`) that always deploys to dev and always name your PR branches `feature`.

**Pull request merged**